hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -220,6 +210,28 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
     }
   }
 
+  @Override
+  public TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
+    // validate first if view exists with the same name or not.
+    try {
+      String database = identifier.namespace().level(0);
+      String tableName = identifier.name();
+      Optional<Table> table = HiveCatalogUtil.loadHmsTableIfPresent(clients, database, tableName);
+      if (table.isPresent()
+          && table.get().getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
+        throw new org.apache.iceberg.exceptions.AlreadyExistsException(
+            "View with same name already exists: %s.%s", database, tableName);
+      }
```

Comment:
Is this only for View-s?
If so, I think this is a serious miss-use ot the method here.